### PR TITLE
Fix/16705 private queries can be accessed by other user via url

### PR DESCRIPTION
**Pull request by tessi**
Fixes: https://community.openproject.org/work_packages/16705

Private queries could be accessed by people, who are able to view work packages in a project.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://fix-16705-private-queries-can-be-accessed-by-other-user-via-url.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/policies/query_policy.rb [hunk 1]
```diff
@@ -71,6 +77,14 @@ def depublicize_allowed?(query)
     manage_public_queries_allowed?(query)
   end
 
+  def view_work_packages_allowed?(query)
```
Comment (ulferts) [inline app/policies/query_policy.rb:80]: @tessi, one of the benefits of using this pattern is that it is easier to write test. So why are there none?

**Comment by ulferts** [comment]
Looks good except for the missing specs.

File: app/policies/query_policy.rb (hunk 1 above)
Comment (tessi) [inline app/policies/query_policy.rb:80]: It's only indirectly tested by the other spec. I'll add specific policy tests.

**Comment by tessi** [comment]
@ulferts you'll find the specs you wished for, now

**Comment by tessi** [comment]
I know the query_loading spec is failing. A fix is in progress.

**Pull request merged**